src/main/java/com/google/cloud/spanner/pgadapter/ConnectionHandler.java

```
@@ -67,22 +69,40 @@
   private static AtomicInteger incrementingConnectionId = new AtomicInteger(0);
   private ConnectionMetadata connectionMetadata;
   private WireMessage message;
-  private Connection jdbcConnection;
+  private Connection spannerConnection;
 
-  ConnectionHandler(ProxyServer server, Socket socket) throws SQLException {
+  ConnectionHandler(ProxyServer server, Socket socket) {
     super("ConnectionHandler-" + CONNECTION_HANDLER_ID_GENERATOR.incrementAndGet());
     this.server = server;
     this.socket = socket;
     this.secret = new SecureRandom().nextInt();
-    this.jdbcConnection =
-        DriverManager.getConnection(server.getOptions().getConnectionURL(), server.getProperties());
+    String uri = server.getOptions().getConnectionURL();
+    if (uri.startsWith("jdbc:")) {
```

This allows users to still specify a normal JDBC connection string in the options when starting PGAdapter, meaning that this change will not break any existing configurations in use by customers.

---

do we need to support this ?

Yes I understand its a breaking change, but we can throw a pretty error to explain they need to drop that "jdbc" prefix. I rather not support something that doesn't make sense here and leave confusing tech debt

---

To make sure customers dont break, this change should be released with a major version change if we were GA. Though our PG releases AFAIR are still beta versioning (i.e. 0.*.*), so we should have the changelog specify this breaking change

---

In addition to not breaking any existing configurations, I think it is better to support the `jdbc:` prefix in the connection string. That makes the documentation and user experience simpler, as we can just simply state something like this in the documentation: 

> Use a JDBC connection URL to specify additional connection properties that PGAdapter should use for the connection to Cloud Spanner. The documentation for the JDBC connection URL can be found here: https://cloud.google.com/spanner/docs/use-oss-jdbc

That way we don't have to confuse users with internals about the Connection API.

---

Curious: Is that allowed in pure PostgreSQL databases?

---

This URL references the Cloud Spanner / Spangres database on the backend that PGAdapter should use. And yes; this is allowed for Spangres databases.

It's not allowed for real PostgreSQL databases, but that is not what this connection URL is for.